#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_no_debug_postfix.dpatch by Toby Schneider <tes.aubergine@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Do not add a debug postfix, as we get the debug libraries by applying 
## DP: dh_strip. As we want to generate a -dbg package, we need to build Witty 
## DP: with CMAKE_BUILD_TYPE=Debug. The CMakeLists.txt files are telling CMake 
## DP: to append a 'd' (a-la Visual C++) to the debug libraries but we don't 
## DP: need (or want, either) to have that suffix. Sadly, 
## DP: -DVARIABLE:TYPE=VALUE won't overrides SET_TARGET_PROPERTIES, therefore 
## DP: making necessary to patch the CMakeLists.txt project files.

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' witty-3.3.11+dfsg~/src/CMakeLists.txt witty-3.3.11+dfsg/src/CMakeLists.txt
--- witty-3.3.11+dfsg~/src/CMakeLists.txt	2018-08-22 05:46:43.000000000 -0400
+++ witty-3.3.11+dfsg/src/CMakeLists.txt	2018-10-22 15:01:44.191379629 -0400
@@ -490,7 +490,6 @@
   PROPERTIES
     VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
     SOVERSION ${WTTEST_SOVERSION}
-    DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
   )
 ELSE(ENABLE_LIBWTTEST)
   MESSAGE("** Not building libwttest")
@@ -588,7 +587,6 @@
   POST_INSTALL_SCRIPT ${WT_BINARY_DIR}/WtInstall.cmake
   VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
   SOVERSION ${WT_SOVERSION}
-  DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
 )
 
 IF(MSVC)
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' witty-3.3.11+dfsg~/src/Wt/Dbo/CMakeLists.txt witty-3.3.11+dfsg/src/Wt/Dbo/CMakeLists.txt
--- witty-3.3.11+dfsg~/src/Wt/Dbo/CMakeLists.txt	2018-08-22 05:46:43.000000000 -0400
+++ witty-3.3.11+dfsg/src/Wt/Dbo/CMakeLists.txt	2018-10-22 15:01:44.211379804 -0400
@@ -59,7 +59,6 @@
   PROPERTIES
     VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
     SOVERSION ${WTDBO_SOVERSION}
-    DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
   )
   IF(MSVC)
     SET_TARGET_PROPERTIES(
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' witty-3.3.11+dfsg~/src/Wt/Dbo/backend/CMakeLists.txt witty-3.3.11+dfsg/src/Wt/Dbo/backend/CMakeLists.txt
--- witty-3.3.11+dfsg~/src/Wt/Dbo/backend/CMakeLists.txt	2018-08-22 05:46:43.000000000 -0400
+++ witty-3.3.11+dfsg/src/Wt/Dbo/backend/CMakeLists.txt	2018-10-22 15:01:44.215379839 -0400
@@ -77,7 +77,6 @@
   PROPERTIES
     VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
     SOVERSION ${WTDBOSQLITE3_SOVERSION} 
-    DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
     )
 
   IF(MSVC)
@@ -139,7 +138,6 @@
   PROPERTIES
     VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
     SOVERSION ${WTDBOPOSTGRES_SOVERSION}
-    DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
   )
 
   INSTALL_FILES(/include/Wt/Dbo/backend "^Postgres$")
@@ -223,7 +221,6 @@
   PROPERTIES
     VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
     SOVERSION ${WTDBOFIREBIRD_SOVERSION}
-    DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
   )
 
   INSTALL_FILES(/include/Wt/Dbo/backend "^Firebird$")
@@ -279,7 +276,6 @@
   PROPERTIES
     VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
     SOVERSION ${WTDBOMYSQL_SOVERSION}
-    DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
   )
   INSTALL_FILES(/include/Wt/Dbo/backend "^MySQL$")
   INSTALL_FILES(/include/Wt/Dbo/backend "^.*MySQL.*h$")
@@ -336,7 +332,6 @@
   PROPERTIES
     VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
     SOVERSION ${WTDBOMSSQLSERVER_SOVERSION}
-    DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
     )
 
   IF(MSVC)
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' witty-3.3.11+dfsg~/src/Wt/Ext/CMakeLists.txt witty-3.3.11+dfsg/src/Wt/Ext/CMakeLists.txt
--- witty-3.3.11+dfsg~/src/Wt/Ext/CMakeLists.txt	2018-08-22 05:46:43.000000000 -0400
+++ witty-3.3.11+dfsg/src/Wt/Ext/CMakeLists.txt	2018-10-22 15:01:44.203379734 -0400
@@ -52,7 +52,6 @@
 PROPERTIES
   VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
   SOVERSION ${WTEXT_SOVERSION}
-  DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
 )
 
 INSTALL_FILES(/include/Wt/Ext "^[^.]..+[^Ch~]$")
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' witty-3.3.11+dfsg~/src/fcgi/CMakeLists.txt witty-3.3.11+dfsg/src/fcgi/CMakeLists.txt
--- witty-3.3.11+dfsg~/src/fcgi/CMakeLists.txt	2018-08-22 05:46:43.000000000 -0400
+++ witty-3.3.11+dfsg/src/fcgi/CMakeLists.txt	2018-10-22 15:01:44.227379944 -0400
@@ -24,7 +24,6 @@
     PROPERTIES
       VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
       SOVERSION ${WTFCGI_SOVERSION}
-      DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
     )
 
     INSTALL(TARGETS wtfcgi
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' witty-3.3.11+dfsg~/src/http/CMakeLists.txt witty-3.3.11+dfsg/src/http/CMakeLists.txt
--- witty-3.3.11+dfsg~/src/http/CMakeLists.txt	2018-08-22 05:46:43.000000000 -0400
+++ witty-3.3.11+dfsg/src/http/CMakeLists.txt	2018-10-22 15:01:44.191379629 -0400
@@ -110,7 +110,6 @@
   PROPERTIES
     VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
     SOVERSION ${WTHTTP_SOVERSION}
-    DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
   )
 
   IF(MSVC)
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' witty-3.3.11+dfsg~/src/isapi/CMakeLists.txt witty-3.3.11+dfsg/src/isapi/CMakeLists.txt
--- witty-3.3.11+dfsg~/src/isapi/CMakeLists.txt	2018-08-22 05:46:43.000000000 -0400
+++ witty-3.3.11+dfsg/src/isapi/CMakeLists.txt	2018-10-22 15:01:44.195379663 -0400
@@ -24,7 +24,6 @@
     PROPERTIES
       VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
       SOVERSION ${WTISAPI_SOVERSION}
-      DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
     )
     IF(MSVC)
       SET_TARGET_PROPERTIES(
